client/lib/gsuite/new-users.ts

```diff
@@ -86,10 +85,38 @@ const validateOverallEmailAgainstExistingEmails = (
 			: mailBoxError,
 } );
 
+const clearPreviousErrors = ( users: GSuiteNewUser[] ) => {
+	return users.map( user => mapValues( user, field => removePreviousErrors( field ) ) );
+};
+
+const validateNewUserMailboxIsUnique = (
+	{ value: mailBox, error: previousError }: GSuiteNewUserField,
+	mailboxesByCount: { [mailbox: string]: number }
+) => ( {
+	value: mailBox,
+	error:
+		mailboxesByCount[ mailBox ] > 1
+			? i18n.translate( 'Please use a unique mailbox for each user.' )
+			: previousError,
+} );
+
+const validateNewUsersAreUnique = ( users: GSuiteNewUser[] ) => {
+	const mailboxesByCount: { [mailbox: string]: number } = countBy(
+		users.map( ( { mailBox: { value: mailBox } } ) => mailBox )
```

Comment:
`countBy( users, 'mailBox.value' )` should do the same thing and is more concise.